vllm/attention/backends/xformers.py

```diff
@@ -86,20 +191,24 @@ def forward(
         key = key.view(-1, self.num_kv_heads, self.head_size)
         value = value.view(-1, self.num_kv_heads, self.head_size)
 
-        # Reshape the keys and values and store them in the cache.
-        # If key_cache and value_cache are not provided, the new key and value
-        # vectors will not be cached. This happens during the initial memory
-        # profiling run.
-        if key_cache is not None and value_cache is not None:
-            PagedAttentionImpl.reshape_and_cache(key, value, key_cache,
-                                                 value_cache, input_metadata)
+        if kv_cache is not None:
+            key_cache, value_cache = PagedAttention.split_kv_cache(
+                kv_cache, self.num_kv_heads, self.head_size)
+
+            # Reshape the input keys and values and store them in the cache.
+            # If kv_cache is not provided, the new key and value tensors are
+            # not cached. This happens during the initial memory profiling run.
+            PagedAttention.reshape_and_cache(key, value, key_cache,
```

Comment:
One random thought. Let's rename `reshape_and_cache` to something like `write_to_paged_cache`? I feel like the name `reshape_and_cache` is confusing. Also for flashinfer kernel, we probably will not need to do "reshape".

Reply:
`reshape_and_cache` is a method of `PagedAttention`, and is not part any abstract class. So, for FlashInfer, we don't have to use this name. That being said, I agree that `write_to_paged_cache` is easier to understand. Fixed the name.